### PR TITLE
Westmelton2 GeoMag equipment update

### PR DESCRIPTION
**Pull request by TurnerDave**
Updated metadata for Fluxgate sensor and Centaur datalogger at the new West Melton 2 GeoMag site

- [x] network/stations.csv (geomag has network code = GM)
- [x] network/sites.csv (geomag has channel code = 50 for the fluxgate and 51 for the proton)
- [x] calibrations/fluxgate : a text file with Danish Met Insitute calibration results for individual fluxgate sensor and driver 
- [x] assets/dataloggers.csv
- [x] assets/recorders.csv  _comment: N/A_
- [x] assets/sensors.csv  _comment: Existing sensor redeployed_
- [x] install/calibrations.csv   _comment: Existing sensor redeployed_
- [x] install/channels.csv (check that model is there)
- [x] install/components.csv (check that model and response file is there)
- [x] install/connections.csv
- [x] install/dataloggers.csv
- [x] install/gains.csv
- [x] install/recorders.csv   _comment: N/A_
- [x] install/sensors.csv
- [x] install/streams.csv
- [x] /files (check model is there)  _comment: resp/files/sensor_Danish_Meteorological_Institute_FGE.xml exists_

### CODE REVIEW

**Comment by elidana** [comment]
Hei @TurnerDave , this is a new site, and will require a bunch of more metadata to be added on delta before this pull request can be merged.

The complete set of files that need to include geomag metadata is:

- network/stations.csv (geomag has network code = GM)
- network/sites.csv (geomag has channel code = 50 for the fluxgate and 51 for the proton)
- calibrations/fluxgate : a text file with Danish Met Insitute calibration results for individual fluxgate sensor and driver
- assets/dataloggers.csv
- assets/recorders.csv
- assets/sensors.csv
- install/calibrations.csv
- install/channels.csv (check that model is there)
- install/components.csv (check that model and response file is there)
- install/connections.csv
- install/dataloggers.csv
- install/gains.csv
- install/recorders.csv
- install/sensors.csv
- install/streams.csv
- resp/files (check model is there)

**Review by elidana — CHANGES_REQUESTED** [review]
Hei @TurnerDave , this is a new site, and will require a bunch of more metadata to be added on delta before this pull request can be merged.

The complete set of files that need to include geomag metadata is:

- network/stations.csv (geomag has network code = GM)
- network/sites.csv (geomag has channel code = 50 for the fluxgate and 51 for the proton)
- calibrations/fluxgate : a text file with Danish Met Insitute calibration results for individual fluxgate sensor and driver
- assets/dataloggers.csv
- assets/recorders.csv
- assets/sensors.csv
- install/calibrations.csv
- install/channels.csv (check that model is there)
- install/components.csv (check that model and response file is there)
- install/connections.csv
- install/dataloggers.csv
- install/gains.csv
- install/recorders.csv
- install/sensors.csv
- install/streams.csv
- resp/files (check model is there)

**Comment by TurnerDave** [comment]
Thanks @elidana I have no idea what I'm doing, so I really appreciate your guidance. I'll update the field you listed :)

**Comment by TurnerDave** [comment]
@elidana check-list completed and changes made where required. Thank you for your help.

**Comment by TurnerDave** [comment]
Note that I need to confirm the correct  install/gains.csv values following feedback from Mark C.  Will re-submit once updated.

**Comment by TurnerDave** [comment]
Correct gain values updated. PR ready for review. Thanks for your help and guidance @elidana and @ozym

File: network/stations.csv [hunk 1]
```diff
@@ -940,6 +940,7 @@ EXPV,XX,Express,-39.253545373,175.564881983,2030,,WGS84,1995-10-25T03:58:00Z,199
 EYCZ,CB,Eyrewell,-43.42136,172.35544,149.7,,WGS84,2012-03-20T00:00:00Z,2013-12-03T00:00:00Z
 EYRM,GM,Eyrewell Geomagnetic Observatory,-43.42134,172.35544,115,,WGS84,1978-01-01T00:00:00Z,2013-12-15T23:59:59Z
 EYRS,SM,Eyrewell Observatory,-43.42138,172.35544,119,,WGS84,2011-03-25T00:00:00Z,2012-10-15T10:45:00Z
+EY2M,GM,West Melton 2 Geomagnetic Observatory,-43.47384,172.39266,102,,WGS84,2024-04-04T16:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline network/stations.csv:943]: - please change the name to `West Melton Geomagnetic Observatory 2` (or keep as is)
- the height here is 102, and in sites is 88. what is the true one? we shall use the same

File: network/sites.csv [hunk 2]
```diff
@@ -862,6 +862,7 @@ EWZ,10,-43.510031722,170.852603355,650,,WGS84,Unknown,1991-03-21T03:43:00Z,2001-
 EXPV,10,-39.253545373,175.564881983,2030,,WGS84,Unknown,1995-10-25T03:58:00Z,1995-11-24T06:22:00Z
 EYCZ,10,-43.42136,172.35544,149.7,,WGS84,External GPS Device,2012-03-20T21:00:01Z,2013-12-02T22:00:00Z
 EYRS,20,-43.42138,172.35544,119,,WGS84,External GPS Device,2011-03-25T05:00:00Z,2012-10-15T10:45:00Z
+EY2M,50,-43.47384,172.39266,88,,WGS84,Unknown,2024-04-04T16:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline network/sites.csv:865]: - please check the height here, need to be consistent with the info in stations.csv

File: install/connections.csv [hunk 3]
```diff
@@ -197,6 +197,7 @@ ETVZ,30,East Tongariro,,3,2014-03-11T00:01:00Z,9999-01-01T00:00:00Z
 EWZ,10,Erewhon,Short Period,,1991-03-21T03:43:00Z,2001-07-02T14:38:00Z
 EXPV,10,Express,Short Period,,1995-10-25T03:58:00Z,1995-11-24T06:22:00Z
 EYCZ,10,Eyrewell,,,2012-03-20T21:00:01Z,2013-12-02T22:00:00Z
+EY2M,50,West Melton Geomagnetic Observatory,,,2024-04-04T16:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline install/connections.csv:200]: please update station name to `West Melton Geomagnetic Observatory 2` or `West Melton 2 Geomagnetic Observatory`

File: install/dataloggers.csv [hunk 4]
```diff
@@ -1032,6 +1032,7 @@ Nanometrics,Centaur CTR4-6S,9917,Okains Bay,,2023-05-23T01:14:00Z,9999-01-01T00:
 Nanometrics,Centaur CTR4-6S,9918,Rangatira Station,,2023-08-03T11:24:00Z,9999-01-01T00:00:00Z
 Nanometrics,Centaur CTR4-6S,9920,Kaweka Forest,,2023-07-18T03:24:00Z,9999-01-01T00:00:00Z
 Nanometrics,Centaur CTR4-6S,9921,Tory Channel,,2023-06-19T00:07:00Z,9999-01-01T00:00:00Z
+Nanometrics,Centaur CTR4-6S,10922,West Melton 2,,2024-04-04T16:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline install/dataloggers.csv:1035]: please update station name to `West Melton Geomagnetic Observatory 2` or `West Melton 2 Geomagnetic Observatory`

File: install/gains.csv [hunk 5]
```diff
@@ -17,6 +17,9 @@ CPXB,27,,,0.1,,,2010-05-01T00:00:00Z,2015-05-07T02:00:00Z
 CPXB,28,,,0.1,,,2010-05-01T00:00:00Z,2015-05-07T02:00:00Z
 CPXB,29,,,0.1,,,2010-05-01T00:00:00Z,2015-05-07T02:00:00Z
 CPXB,2A,,,0.1,,,2010-05-01T00:00:00Z,2015-05-07T02:00:00Z
+EY2M,50,,X,,0x67,,2024-04-042T16:00:00Z,9999-01-01T00:00:00Z
+EY2M,50,,Y,,0x00,,2024-04-042T16:00:00Z,9999-01-01T00:00:00Z
+EY2M,50,,Z,,0x8c,,2024-04-042T16:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline install/gains.csv:22]: to review these, can you please provide a photo of the instrument setup in the equipment change ticket?

File: install/sensors.csv [hunk 6]
```diff
@@ -39,6 +39,7 @@ City Technology,T3H H2S sensor,01.44865262 112,NA002,06,0,,0,0,0,0,0,0,2024-03-2
 City Technology,T3STF SO2 sensor,01.45929880 013,TO003,13,0,,0,0,0,0,0,0,2024-03-01T00:20:00Z,9999-01-01T00:00:00Z
 City Technology,T3STF SO2 sensor,01.45929894 013,NA002,05,0,,0,0,0,0,0,0,2024-03-21T02:10:00Z,9999-01-01T00:00:00Z
 Danish Meteorological Institute,FGE,S0257-E0299,SBAM,50,0,,0,0,0,0,0,0,2009-03-02T00:00:00Z,2023-10-22T23:59:00Z
+Danish Meteorological Institute,FGE,S0257-E0299,EY2M,50,0,,0,0,0,0,0,0,2024-04-04T00:00:00Z,9999-01-01T00:00:00Z
```
Comment (elidana) [inline install/sensors.csv:42]: please update start time to be consistent with other entries in install files
`2024-04-04T16:00:00Z`

**Review by elidana — CHANGES_REQUESTED** [review]
well done on wrangling this! some more comments and tweaks

A couple of general comments (in addition to the more detailed in line ones) 
- is there a calibration file for this sensor? If so, can we please add it to the `calibration/fluxgates` folder. This is usually a copy of the calibration PDF received by the Danish Met Institute, file name has the same serial number used here
- In general, the station name needs to be consistent across different files. I propose to use 
`West Melton Geomagnetic Observatory 2` or `West Melton 2 Geomagnetic Observatory` everywhere

see also other more detailed comments

File: install/connections.csv (hunk 3 above)
Comment (TurnerDave) [reply install/connections.csv]: Amended

File: network/stations.csv (hunk 1 above)
Comment (TurnerDave) [reply network/stations.csv]: I think we'll stick with West Melton 2 Geomagnetic Observatory - I'll amend other locations accordingly.
I copied height from existing West Melton site - I'll amend both to the correct height of 88 in sites.csv

File: network/sites.csv (hunk 2 above)
Comment (TurnerDave) [reply network/sites.csv:865]: This is the correct value

File: install/dataloggers.csv (hunk 4 above)
Comment (TurnerDave) [reply install/dataloggers.csv]: Done

File: install/gains.csv (hunk 5 above)
Comment (TurnerDave) [reply install/gains.csv]: Amended and added photo of gains to ticket https://github.com/GeoNet/tickets/issues/15482

File: install/sensors.csv (hunk 6 above)
Comment (TurnerDave) [reply install/sensors.csv]: Amended

**Comment by TurnerDave** [comment]
@elidana We're getting there.  All points above addressed and ready for review.  Thanks for your patience :)

**Comment by elidana** [comment]
- thanks for addressing all comments! 
- calibration file: APOLOGIES! i did not realised that the sensor was already in use in SBAM in the past. Calibration file was already available in delta (https://github.com/GeoNet/delta/blob/main/calibration/fluxgates/FGE_S0257.const.md) so can you please remove the uploaded PDF? (for calibrations, we usually reformat the PDF provided by the Danish institute, so that the info is a bit more structured). sorry for the unnecessary extra work
- some tests are still failing, will check how to fix and comment here. 
- thanks for adding info on gains in the internal ticket!

**Comment by TurnerDave** [comment]
@elidana I should have questioned adding the pdf, as the other sensors didn't have their calibration certs uploaded.  I now have a problem that I can't delete the pdf from the pull request.  The option to delete the file is greyed-out.  I can delete the entire directory, but not that file inside of it.  Any ideas?

**Comment by elidana** [comment]
ok, I did some tweaks on this pr
- deleted the pdf (apologies for the confusing message and for missing the info were there already)
- moved entries to pass the tests (they need to be in alpha-numerical orders in all files)
- squashed all commits into one

all tests are now passing

**Comment by elidana** [comment]
@CBurton90 or @gva6500 can you please review this from the RIM team perspective?

@ozym , this is a new site. I can see already miniseed data poppinp up in the archive, but would be great if you can double check that this is complete!

**Review by ozym — APPROVED** [review]
passes the tests and has the gain settings.

**Review by CBurton90 — DISMISSED** [review]
Looking good, thanks for your efforts here @TurnerDave

**Comment by elidana** [comment]
install times updated to reflect first data in the archive
```
FDSN:NZ_EY2M_50_L_F_X, 2, 512, 412 samples, 1 Hz, 2024-04-09T02:29:55.000000Z (100)
FDSN:NZ_EY2M_50_L_F_X, 2, 512, 412 samples, 1 Hz, 2024-04-09T02:36:47.000000Z (100)
FDSN:NZ_EY2M_50_L_F_X, 2, 512, 412 samples, 1 Hz, 2024-04-09T02:43:39.000000Z (100)
FDSN:NZ_EY2M_50_L_F_X, 2, 512, 412 samples, 1 Hz, 2024-04-09T02:50:31.000000Z (100)
FDSN:NZ_EY2M_50_L_F_X, 2, 512, 412 samples, 1 Hz, 2024-04-09T02:57:23.000000Z (100)
```

**Review by elidana — APPROVED** [review]

**Comment by tanjapetersen** [comment]
Thanks so much for all your hard work to get this done!!

**Pull request merged**